### ISSUE TITLE
[FEATURE REQUEST] Allow to disable the web UI

### ISSUE THREAD

**Issue by williamdes** [lldap/lldap#913]
**Motivation**

I have multiple servers, they do not each need the web UI.

**Describe the solution you'd like**

Allow to set to `None` the web port or host to disable the web server bind.

https://github.com/lldap/lldap/blob/4f050cded5ccdbb71898d9611adbcc45533c0833/server/src/main.rs#L170

**Describe alternatives you've considered**

Not providing the web UI files.

**Comment by nitnelave** [comment]
Hi! I feel like this is better solved using firewall rules. I don't think it's necessary to add an option to disable the web UI. What if you wanted to serve the web UI on 2 ports, would you want an option for that? This would be better solved with a reverse proxy in front.

**Comment by nitnelave** [comment]
Note that disabling the http port would also disable the GraphQL API

**Comment by williamdes** [comment]
> Hi! I feel like this is better solved using firewall rules. I don't think it's necessary to add an option to disable the web UI. What if you wanted to serve the web UI on 2 ports, would you want an option for that? This would be better solved with a reverse proxy in front.

Well, yes and no.
I really want that the code is disabled and not to have a warning when the service boots.

**Comment by williamdes** [comment]
Could you re open and consider this feature?
I can make a PR

**Comment by nitnelave** [comment]
I'm not sure I understand, what warning are you seeing?

**Comment by williamdes** [comment]
> I'm not sure I understand, what warning are you seeing?

Delete the app folder and restart. It will complain not having it

**Comment by nitnelave** [comment]
Yes, that's not what I'm suggesting. I'm suggesting a firewall or a reverse proxy in front of it to disable traffic to the web Gui

**Comment by williamdes** [comment]
> Yes, that's not what I'm suggesting. I'm suggesting a firewall or a reverse proxy in front of it to disable traffic to the web Gui

yes, but that's what I am reporting ;)
First the warning should go away if I do not want to have the web ui running and binding my port.

**Comment by williamdes** [comment]
> I hear you :) I'm trying to reduce the complexity of LLDAP, especially to support unusual cases. How bad would it be to have the static files, bind to a port (HTTP) and disable access to that port in the firewall or proxy? Are you worried about the resource usage? In terms of CPU, disk space, number of free ports? I don't have all your constraints.

I am trying to use rust tooling because the mentality is more to not use what you do not need. And I do not need the Web UI at all.

That removes security concerns about the web API and the files itself.

Just do not have what you do not need. 

That said, off topic, but Idid not find a way to load custom schemas. And sadly I am back to openldap. 😢

**Comment by nitnelave** [comment]
We can't load ldif schemas, but we have custom user defined attributes which allow you to build your schema. There are some limitations on the filters that you can use, and they don't have a great web UI yet, though. 

I'm not convinced by the concerns of resources (it's very cheap to have an interface that you listen to) or security (the main problem would be the one server where you do expose the web UI).
